projects/RabbitMQ.Client/client/impl/ModelBase.cs

```diff
@@ -910,7 +916,44 @@ public void BasicPublish<TProperties>(string exchange, string routingKey, in TPr
             }
 
             var cmd = new BasicPublish(exchange, routingKey, mandatory, default);
-            ModelSend(in cmd, in basicProperties, body);
+            using (Activity sendActivity = RabbitMQActivitySource.Send(cmd, body.Length))
+            {
+                if (sendActivity != null)
+                {
+                    if (sendActivity.IsAllDataRequested)
+                    {
+                        if (!string.IsNullOrEmpty(basicProperties.CorrelationId))
+                        {
+                            sendActivity.AddTag("messaging.conversation_id", basicProperties.CorrelationId);
+                        }
+                    }
+
+                    // This activity is marked as recorded, so let's propagate the trace and span ids.
```

Comment:
the expected behavior would be to align with : https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/MicroserviceExample/Utils/Messaging

Reply:
Thank you for pointing this out. I'll make it behave the same way :)